bbq/compiler/compiler.go

```diff
@@ -52,8 +52,17 @@ type Compiler[E, T any] struct {
 	usedImportedGlobals []*global
 	controlFlows        []controlFlow
 	currentControlFlow  *controlFlow
+	returns             []returns
+	currentReturn       *returns
 	staticTypes         []T
 
+	// postConditionsIndices keeps track of where the post conditions start (i.e: index of the statement in the block),
+	// for each function. This is used by the compiler to patch the jumps for return statements.
+	postConditionsIndices map[ast.Declaration]int
+
+	// postConditionsIndex is the statement-index of the post-conditions for the current function.
+	postConditionsIndex int
```

Comment:
Does this assume the post conditions were added by the desugaring/rewrite to the end of the function's block?

Reply:
Correct. This `postConditionsIndices` is populated by/during the desugar/rewrite

Reply:
Makes sense 👍 

Could you please add this detail to the comment? It seems essential to understanding where the information comes from

Reply:
Improved the comment here: https://github.com/onflow/cadence/pull/3816/commits/49b91a2a4f9916b9d6d6ce5d23c3c6debe0ab9ff